network/src/protocols/direct_send/mod.rs

```diff
@@ -292,8 +292,12 @@ where
             }
         };
 
-        substream_queue_tx.send(msg.mdata).await.map_err(|e| {
-            self.message_queues.remove(&(peer_id, protocol));
+        substream_queue_tx.try_send(msg.mdata).map_err(|e| {
+            // If the channel is full, simply drop the message on the floor;
```

Comment:
will be good to log a WARN when we hit this, and maybe add a metric too.

Reply:
right, we actually do log a warn and bump the dropped message counter in the parent function `handle_direct_send_request`, so it should be good.

Reply:
the function below (`handle_direct_send_request`) handles this already :)